First submission of TPS & CTPS Checker connector

WilkoUK opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [X] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [ ] I attest that the connector works and I verified by deploying and testing all the operations. 
- [X] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [X] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [ ] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [X] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [ ] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [ ] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [ ] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [ ] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md. 


Testing & Validation tools described in the MS documentation either did not work or the documentation  did not match reality.

WilkoUK committed:
Add New Connector Files

New Certified Connector assets for TPS CTPS Checker

WilkoUK committed:
Add icon file to repository

WilkoUK committed:
Add icon file to repository

WilkoUK committed:
Update Readme.md

WilkoUK committed:
Merge branch 'dev' of https://github.com/WilkoUK/PowerPlatformConnectors into dev

WilkoUK committed:
Update icon.png

WilkoUK committed:
Change to iconBrandColor

WilkoUK commented:
@microsoft-github-policy-service agree [company=“Wilkins Computing Services Limited"]
@microsoft-github-policy-service agree
@microsoft-github-policy-service agree company="Microsoft"

Paul Wilkins

***@***.***



> On 16 Apr 2024, at 12:34, microsoft-github-policy-service[bot] ***@***.***> wrote:
> 
> 
> @WilkoUK <https://github.com/WilkoUK> please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> @microsoft-github-policy-service agree [company=“Wilkins Computing Services Limited"]
> Options:
> 
> (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> @microsoft-github-policy-service agree
> (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> @microsoft-github-policy-service agree company="Microsoft"
> Contributor License Agreement
>  <http://www.opensource.org/>
> —
> Reply to this email directly, view it on GitHub <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#issuecomment-2058877731>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BHXIKUE63GJ5TMDXFZ3UWQ3Y5UEF7AVCNFSM6AAAAABGJFCPV6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANJYHA3TONZTGE>.
> You are receiving this because you were mentioned.
>

vmanoharas commented on certified-connectors/TPS & CTPS Checker/Readme.md:
```diff
@@ -0,0 +1,20 @@
+# TPS & CTPS Checker
+TPS & CTPS Checker allows you to query the UK Telephone Preference Service for a given UK Telephone Number.
```
please format the readme.md file according to [the template](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md). All of the required headings should be present.

vmanoharas commented on certified-connectors/TPS & CTPS Checker/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,167 @@
+            "name": "tel",
+            "type": "string",
+            "x-ms-summary": "UK Telephone Number",
+            "description": "The UK basewd telephone number to query"
```
Kindly correct the typo "basewd"

vmanoharas reviewed (commented):
Hello @WilkoUK, 

Kindly resolve review comments. Additionally, Update the logo of with 1:1 dimensions within a range of 230 × 230 pixels (no rounded edges).

WilkoUK committed:
Updated readme.md

WilkoUK committed:
Correction to typo

WilkoUK commented:
Good Morning @vmanoharas,

I have made the changes that you requested and pushed them to my clone repository, I am new to this so I am not sure if I need to do anything else for you to collect these changes.

With regard to the icon. I am not sure what you are requesting. Do you require the icon to be 230x230? Documentation says within 100x100 ~ 230x230. 

Many Thanks.

Paul Wilkins

***@***.***



> On 17 Apr 2024, at 10:46, vmanoharas ***@***.***> wrote:
> 
> 
> @vmanoharas commented on this pull request.
> 
> Hello @WilkoUK <https://github.com/WilkoUK>,
> 
> Kindly resolve review comments. Additionally, Update the logo of with 1:1 dimensions within a range of 230 × 230 pixels (no rounded edges).
> 
> In certified-connectors/TPS & CTPS Checker/Readme.md <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#discussion_r1568541816>:
> 
> > @@ -0,0 +1,20 @@
> +# TPS & CTPS Checker
> +TPS & CTPS Checker allows you to query the UK Telephone Preference Service for a given UK Telephone Number.
> please format the readme.md file according to the template <https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md>. All of the required headings should be present.
> 
> In certified-connectors/TPS & CTPS Checker/apiDefinition.swagger.json <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#discussion_r1568547404>:
> 
> > +        "summary": "Query the TPS/CTPS status of UK based telephone number",
> +        "operationId": "Query",
> +        "parameters": [
> +          {
> +            "in": "query",
> +            "name": "api_key",
> +            "type": "string",
> +            "x-ms-summary": "API Key",
> +            "description": "Your API Key provided by your tpsmydata account"
> +          },
> +          {
> +            "in": "query",
> +            "name": "tel",
> +            "type": "string",
> +            "x-ms-summary": "UK Telephone Number",
> +            "description": "The UK basewd telephone number to query"
> Kindly correct the typo "basewd"
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#pullrequestreview-2005543186>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BHXIKUA5YM4RQBJ65OJJLXTY5ZAFRAVCNFSM6AAAAABGJFCPV6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMBVGU2DGMJYGY>.
> You are receiving this because you were mentioned.
>

WilkoUK committed:
Update icon.png

vmanoharas commented:
> Good Morning @vmanoharas, I have made the changes that you requested and pushed them to my clone repository, I am new to this so I am not sure if I need to do anything else for you to collect these changes. With regard to the icon. I am not sure what you are requesting. Do you require the icon to be 230x230? Documentation says within 100x100 ~ 230x230. Many Thanks. Paul Wilkins ***@***.***
> […](#)
> On 17 Apr 2024, at 10:46, vmanoharas ***@***.***> wrote: @vmanoharas commented on this pull request. Hello @WilkoUK <https://github.com/WilkoUK>, Kindly resolve review comments. Additionally, Update the logo of with 1:1 dimensions within a range of 230 × 230 pixels (no rounded edges). In certified-connectors/TPS & CTPS Checker/Readme.md <[#3376 (comment)](https://github.com/microsoft/PowerPlatformConnectors/pull/3376#discussion_r1568541816)>: > @@ -0,0 +1,20 @@ +# TPS & CTPS Checker +TPS & CTPS Checker allows you to query the UK Telephone Preference Service for a given UK Telephone Number. please format the readme.md file according to the template <https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md>. All of the required headings should be present. In certified-connectors/TPS & CTPS Checker/apiDefinition.swagger.json <[#3376 (comment)](https://github.com/microsoft/PowerPlatformConnectors/pull/3376#discussion_r1568547404)>: > + "summary": "Query the TPS/CTPS status of UK based telephone number", + "operationId": "Query", + "parameters": [ + { + "in": "query", + "name": "api_key", + "type": "string", + "x-ms-summary": "API Key", + "description": "Your API Key provided by your tpsmydata account" + }, + { + "in": "query", + "name": "tel", + "type": "string", + "x-ms-summary": "UK Telephone Number", + "description": "The UK basewd telephone number to query" Kindly correct the typo "basewd" — Reply to this email directly, view it on GitHub <[#3376 (review)](https://github.com/microsoft/PowerPlatformConnectors/pull/3376#pullrequestreview-2005543186)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BHXIKUA5YM4RQBJ65OJJLXTY5ZAFRAVCNFSM6AAAAABGJFCPV6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMBVGU2DGMJYGY>. You are receiving this because you were mentioned.

Very Good morning @WilkoUK, 

Thank you very much for addressing the comments. Yes  the icon to be 230x230.

WilkoUK committed:
Update icon.png

Replace original 100x100 image with 230x230 image

WilkoUK commented:
Good Morning @vmanoharas,

Thank you for clearing up the confusion. I have now supplied a 230x230 icon and pushed it to my clone repository for your action.

I suggest that the MS documentation <https://learn.microsoft.com/en-us/connectors/custom-connectors/certification-submission#design-an-icon-for-your-connector> relating to icons is updated to reflect the current requirement.

Hope to hear form you again soon.

Very best wishes,
Paul 

***@***.***



> On 17 Apr 2024, at 17:46, vmanoharas ***@***.***> wrote:
> 
> 
> Good Morning @vmanoharas <https://github.com/vmanoharas>, I have made the changes that you requested and pushed them to my clone repository, I am new to this so I am not sure if I need to do anything else for you to collect these changes. With regard to the icon. I am not sure what you are requesting. Do you require the icon to be 230x230? Documentation says within 100x100 ~ 230x230. Many Thanks. Paul Wilkins @.***
> … <x-msg://4/#>
> On 17 Apr 2024, at 10:46, vmanoharas @.***> wrote: @vmanoharas <https://github.com/vmanoharas> commented on this pull request. Hello @WilkoUK <https://github.com/WilkoUK> https://github.com/WilkoUK, Kindly resolve review comments. Additionally, Update the logo of with 1:1 dimensions within a range of 230 × 230 pixels (no rounded edges). In certified-connectors/TPS & CTPS Checker/Readme.md <#3376 (comment) <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#discussion_r1568541816>>: > @@ -0,0 +1,20 @@ +# TPS & CTPS Checker +TPS & CTPS Checker allows you to query the UK Telephone Preference Service for a given UK Telephone Number. please format the readme.md file according to the template https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/certified-connectors/readme.md. All of the required headings should be present. In certified-connectors/TPS & CTPS Checker/apiDefinition.swagger.json <#3376 (comment) <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#discussion_r1568547404>>: > + "summary": "Query the TPS/CTPS status of UK based telephone number", + "operationId": "Query", + "parameters": [ + { + "in": "query", + "name": "api_key", + "type": "string", + "x-ms-summary": "API Key", + "description": "Your API Key provided by your tpsmydata account" + }, + { + "in": "query", + "name": "tel", + "type": "string", + "x-ms-summary": "UK Telephone Number", + "description": "The UK basewd telephone number to query" Kindly correct the typo "basewd" — Reply to this email directly, view it on GitHub <#3376 (review) <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#pullrequestreview-2005543186>>, or unsubscribe https://github.com/notifications/unsubscribe-auth/BHXIKUA5YM4RQBJ65OJJLXTY5ZAFRAVCNFSM6AAAAABGJFCPV6VHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDAMBVGU2DGMJYGY. You are receiving this because you were mentioned.
> 
> Very Good morning @WilkoUK <https://github.com/WilkoUK>,
> 
> Thank you very much for addressing the comments. Yes the icon to be 230x230.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#issuecomment-2061743742>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BHXIKUEPTBYO7O463CJSITDY52RPFAVCNFSM6AAAAABGJFCPV6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANRRG42DGNZUGI>.
> You are receiving this because you were mentioned.
>

WilkoUK committed:
Merge branch 'microsoft:dev' into dev

vmanoharas commented:
Hello @WilkoUK, 

Thank you for your suggestion, we will update the documentation. 

However, Your merge commit has an issue, it has not merged properly, it is conflicting with other connector repos, please refer the apiProperties validation checker errors. Please recommit or close this PR and open a new PR. **Please don't merge it until the review is completed**.

Additionally, please agree CLA.

WilkoUK commented:
Hi :)

I’ve closed the first pull request and re-opened a new one.
I’ve replied to the CLA 
I’m looking at the errors being generated and I haven’t got a clue what the errors relate to. If you could lend a hand, I’d be very grateful.

Best Wishes
Paul W

> On 18 Apr 2024, at 16:48, vmanoharas ***@***.***> wrote:
> 
> 
> Hello @WilkoUK <https://github.com/WilkoUK>,
> 
> Thank you for your suggestion, we will update the documentation.
> 
> However, Your merge commit has an issue, it has not merged properly, it is conflicting with other connector repos, please refer the apiProperties validation checker errors. Please recommit or close this PR and open a new PR. Please don't merge it until the review is completed.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/microsoft/PowerPlatformConnectors/pull/3376#issuecomment-2064315784>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/BHXIKUD3AG737GIPK2L2RZ3Y57TN3AVCNFSM6AAAAABGJFCPV6VHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDANRUGMYTKNZYGQ>.
> You are receiving this because you were mentioned.
>
